### ISSUE TITLE
[ANDROID] Tab Bar - tabBarVisible = false displays grey bar

### ISSUE THREAD

**Issue by brandtnewww** [react-navigation/navigation-ex#165]
### Current Behavior

On Android, when I set tabBarVisible to false, I still get a grey bar at the bottom of the screen.

Implemented like this:

```
<Tab.Screen name={AppNavigatorKeys.profileTab.routeName} component={ProfileScreen} options={{ tabBarVisible: false }} />
```

![tabBar-visible-false](https://user-images.githubusercontent.com/44766428/68523925-e3147800-02bf-11ea-8f70-96077bb94b39.png)

### Expected Behavior

I except the tab bar to be hidden, like in iOS.

### How to reproduce

Set options={{ tabBarVisible: false }} of a Tab.Screen.

### Your Environment

```
     "react-native": "0.61.2",
    "@react-navigation/bottom-tabs": "5.0.0-alpha.20",
    "@react-navigation/core": "5.0.0-alpha.21",
    "@react-navigation/native": "^5.0.0-alpha.15",
    "@react-navigation/stack": "5.0.0-alpha.35",
```

**Comment by satya164** [comment]
I cannot repro this on my device. Someone else told me that they have similar issue on some Android devices (not all) with React Navigation 4. So suspect there is some issue with React Native which is causing this.

**Comment by brandtnewww** [comment]
@satya164 Thanks a lot for investigating so fast! 😊I investigated a little further. Seems like my bug report was not complete because I didn't understand how fast refresh works with code changes and I wanted the report to be a little simpler.

So here's the full bug report. Maybe you can get more out of it.

I define if the tab bar should be displayed based on the screen the user navigates to via the screenOptions props of the whole navigator, not the individual tab option prop of the tab:

```
<Tab.Navigator tabBarOptions={tabBarOptions} screenOptions={(route) => tabScreenOptions(route)}>
    <Tab.Screen name={AppNavigatorKeys.lessonTab.routeName} component={LearnStack} />
    <Tab.Screen name={AppNavigatorKeys.badgeTab.routeName} component={BadgeStack} />
    <Tab.Screen name={AppNavigatorKeys.profileTab.routeName} component={ProfileScreen} />
    <Tab.Screen name={AppNavigatorKeys.moreTab.routeName} component={MoreScreen} /></Tab.Navigator>
```

```
const tabScreenOptions = props => {
  const { name } = props.route
  const { label, color } = tabConfiguration(name)
  return ({
    tabBarIcon: ({ focused }) => {
      return <TabBarIcon name={name} focused={focused} />
    },
    tabBarLabel: ({ focused }) => {
      return <TabBarLabel bold focused={focused} activeTintColor={color}>{label}</TabBarLabel>
    },
    tabBarVisible: getTabBarVisibleState(props.route)
  })
}
```

The function to get the tabBarVisible setting:
```
const getTabBarVisibleState = (route) => {
  let tabBarVisible = true

  if (route.name === AppNavigatorKeys.lessonTab.routeName) {
    if (route.state && route.state.index > 0) {
      tabBarVisible = false
    } else if (route.state && route.state.routes[0].name === AppNavigatorKeys.quizStack.routeName) {
      tabBarVisible = false
    }
    return tabBarVisible
  }

  return tabBarVisible
}
```

**Comment by karanwadhwa** [comment]
@satya164 i have a similar issue where the bottom tab navigator is greyed out on release builds, when i set dev = false while bundling.
`
    "@react-navigation/bottom-tabs": "^5.0.0-alpha.25",
    "@react-navigation/native": "^5.0.0-alpha.18",
    "@react-navigation/stack": "^5.0.0-alpha.43",`

**Comment by satya164** [comment]
We recommend to reorganize your navigators instead of relying on hiding the tab bar. Which would also avoid such issues https://reactnavigation.org/docs/en/next/screen-options-resolution.html#setting-parent-screen-options-based-on-child-navigators-state

**Comment by karanwadhwa** [comment]
it happens to me even when I'm not hiding the tab navigator anywhere. on v5 when i bundle my app with dev=false ( release builds) the bottom tab navigator is always greyed out like in the image above on the very first screen. this issue does not pop up if i bundle my app using dev=true

**Comment by satya164** [comment]
Are you using `react-native-screens`? Are you on latest versions of all packages?

**Comment by karanwadhwa** [comment]
@satya164 yes i am using `react-native-screens` and I'm on latest versions of all packages.
```js
"react-native-screens": "~2.0.0-alpha.12",
"@react-navigation/bottom-tabs": "^5.0.0-alpha.39",
"@react-navigation/drawer": "^5.0.0-alpha.41",
"@react-navigation/native": "^5.0.0-alpha.29",
"@react-navigation/stack": "^5.0.0-alpha.43",
```

**Comment by satya164** [comment]
https://github.com/kmagiera/react-native-screens/issues/287

**Comment by karanwadhwa** [comment]
was able to reproduce it on ios as well on a release build.
https://imgur.com/a/6rnkfZp
the grey bar pushes the bottom tab navigator down which is why I wasnt able to see it on android screens at all.
Tried removing `SafeAreaProvider`- "react-native-safe-area-context" from root App.js component and `SafeAreaView` from my initial page and kept it in on the second page, same issue on both.
